packages/shared/lib/hooks/hooks.ts

```diff
@@ -27,3 +37,86 @@ export const connectionCreated = async (
 export const connectionCreationFailed = async (connection: RecentlyCreatedConnection, provider: string, activityLogId: number | null): Promise<void> => {
     await webhookService.sendAuthUpdate(connection, provider, false, activityLogId);
 };
+
+export const connectionTest = async (
+    provider: string,
+    template: ProviderTemplate,
+    credentials: ApiKeyCredentials | BasicApiCredentials,
+    connectionId: string,
+    providerConfigKey: string,
+    environment_id: number,
+    connection_config: ConnectionConfig,
+    tracer: Tracer
+): Promise<Result<boolean, NangoError>> => {
+    const providerVerification = template?.proxy?.verification;
+
+    if (!providerVerification) {
+        return resultOk(true);
+    }
+    const active = tracer.scope().active();
+    const span = tracer.startSpan(SpanTypes.CONNECTION_TEST, {
+        childOf: active as Span
+    });
+
+    span.setTag('nango.provider', provider).setTag('nango.providerConfigKey', providerConfigKey).setTag('nango.connectionId', connectionId);
```

Comment:
you can use the `addTags` function to specify multiple tags or even better here the tags attributes of the `startSpanOptions`.

```
    const span = tracer.startSpan(SpanTypes.CONNECTION_TEST, {
        childOf: active as Span,
        tags: {
           'foo': bar, ...
        }
    });
```

Reply:
True, updated with https://github.com/NangoHQ/nango/pull/1656/commits/3dd00af870d00d0e326e46eaae13e1e8fed622e8